### ISSUE TITLE
Informer example requires fix: infomer restarted after onerror event doesn't dispatch other events

### ISSUE THREAD

**Issue by faxioman** [kubernetes-client/javascript#795]
**Describe the bug**
The informer example shows a way to deal with errors:
```
informer.on('error', (err: k8s.V1Pod) => {
  console.error(err);
  // Restart informer after 5sec
  setTimeout(() => {
    informer.start();
  }, 5000);
});
```

But when an error occurs (eg. the api server pod is restarted) if you restart the informer with informer.start(), all the events (add, delete, update) are never fired again.
I don't know if this is a desired behaviour and the example is wrong or the example is fine and the informer has a bug.

** Client Version **
0.16.3

** Server Version **
Kubernetes v1.23.3

**To Reproduce**
Execute the informer, restart the api server, observe that informer events are never fired

**Expected behavior**
After an error event and an informer restart, all add/delete/update event for the informer should be fired again

**Environment (please complete the following information):**
 - OS: MacOs (minikube v1.25.2)
 - NodeJS Version v16.14.0
 - Cloud runtime minikube v1.25.2

**Comment by brendandburns** [comment]
That is expected. Events should only be fired for newly added things (since presumably you already received events for the earlier things).

This should be consistent with the way informers work in other languages also.

**Comment by faxioman** [comment]
Yes, this behaviour is clear… but after informer.start inside an onerror handler, if I remove a k8s resource and recreate that resource the events delete/add are never fired.

**Comment by faxioman** [comment]
@brendandburns ok, I'll check if an update event is raised.
Anyway, I see that if I simply remove a crd (after an informer restart) the event delete is never fired.

**Comment by brendandburns** [comment]
Hrm, that seems like a bug. Once the informer is restarted, if you delete something, you should see the delete event.

Can you provide more details on the repro (or sample code)

thanks

**Comment by brendandburns** [comment]
Friendly ping for more details if you have them.

Thanks!

**Comment by paulorlyk** [comment]
Not sure if it's related, but I observed a similar issue when removing CRD of watched objects. From what I understand (version 0.16.3, file cache.ts), after informer restarts (and it will try to), resourceVersion is never cleared. In my case, after CRD is restored, informer just keeps restarting with no delay and floods API server with requests. No more events from informer after that happens.

Maybe it's possible that resourceVersion is also somehow invalidated during restart of the API server...

I think the issue I described was fixed in [#781](https://github.com/kubernetes-client/javascript/pull/781), but as of now it's not yet published to npm.

As a quick and dirty fix, I had to create a new instance of informer each time "error" event is fired. This way there is still a small possibility that informer might halt and catch fire if CRD was restored before informer had a chance to try to reconnect.
